### ISSUE TITLE
No limit of creatures stacks on a battlefield

### ISSUE THREAD

**Issue by HarkUhm** [vcmi/vcmi#2535]
**Describe the bug**
In OH each player can have no more than 20 stacks of creatures on a battlefield (40 stacks in total) - this number can be reached in ultra rare cases when one (or both) mage on a battlefield summons elementals and have more than around 400 spell points. In VCMI there is no limit. 

**To Reproduce**
1. Make a map for two players, 
2. set both to be playable by human, 
3. give them 99 knowledge
4. give them "summon elemental" spell and "clone" spell
5. start multiplayer game, spawn elementals

**Expected behavior:**
In OH when hero reached the number of 20 stacks and will try to summon more, his hero will receive incorrect info (not exactly this, it's my poor translation) "Your hero summoned earth/water/air/fire elementals and no other elementals can be summoned". It shall inform something like this in English instead: "Your hero already has 20 stacks and can't have more". To see this information, instead of summoning elementals, clone one of your unit other than summoned elemental.

**Version**
 - OS:Windows 10
 - Version: vcmi 1.3

vcmi:
![image](https://github.com/vcmi/vcmi/assets/141812373/fe6a205e-c410-4c89-96e8-be2f8b9a12b3)
HoTA, Polish:
![Bez tytułu](https://github.com/vcmi/vcmi/assets/141812373/26a519ac-5c2e-498d-a74c-57a3d381e4eb)

**Comment by DjWarmonger** [comment]
Not a bug, it's a feature.

**Comment by rilian-la-te** [comment]
Agree. VCMI does not want to replicate all buggy mechanics and exploits which present in OH3.

**Comment by dydzio0614** [comment]
optional toggle can still be made - practical usage of this thing: https://www.youtube.com/watch?v=tuy_ncoBhGM
